Sources/BraintreeVenmo/BTVenmoClient.swift

```diff
@@ -73,36 +73,81 @@ import BraintreeCore
                 self.notifyFailure(with: error, completion: completion)
                 return
             }
-
+            
             guard let configuration else {
                 self.notifyFailure(with: BTVenmoError.fetchConfigurationFailed, completion: completion)
                 return
             }
-
+            
             do {
                 let _ = try self.verifyAppSwitch(with: configuration)
             } catch {
                 self.notifyFailure(with: error, completion: completion)
                 return
             }
-
+            
+            // Merchants are not allowed to collect user addresses unless ECD (Enriched Customer Data) is enabled on the BT Control Panel.
+            if ((request.collectCustomerShippingAddress || request.collectCustomerBillingAddress) && !configuration.isVenmoEnrichedCustomerDataEnabled) {
+                completion(nil, BTVenmoError.enrichedCustomerDataDisabled)
```

Comment:
```suggestion
                notifyFailure(with: BTVenmoError.enrichedCustomerDataDisabled, completion: completion)
```
This will ensure we send the expected analytics events for this client